kymatio/scattering2d/frontend/torch_f.py

```diff
@@ -0,0 +1,178 @@
+
+    def build(self):
+        self.M, self.N = self.shape
+        self.modulus = Modulus()
+        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
+        self.pad = Pad([(self.N_padded - self.N) // 2, (self.N_padded - self.N + 1) // 2, (self.M_padded - self.M) // 2,
+                       (self.M_padded - self.M + 1) // 2], [self.N, self.M], pre_pad=self.pre_pad)
+        self.subsample_fourier = SubsampleFourier()
+        self.filters = self.create_and_register_filters()
+
+    def create_and_register_filters(self):
+        # Create the filters
+        filters = filter_bank(self.M_padded, self.N_padded, self.J, self.L)
+        n = 0
+        self.phi, self.psi = filters['phi'], filters['psi']
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] = torch.from_numpy(self.phi[c])
+                self.register_buffer('tensor' + str(n), self.phi[c])
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = torch.from_numpy(v)
+                    self.register_buffer('tensor' + str(n), self.psi[j][k])
+                    n += 1
+
+    def scattering(self, input):
+        # each time scattering is run, one needs to make sure self.psi and self.phi point to
+        # the correct buffers
+        n=0
+        buffer_dict = dict(self.named_buffers())
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] =  buffer_dict['tensor' + str(n)]
+                n += 1
+
+        for j in range(len(self.psi)):
+            for k, v in self.psi[j].items():
+                if isinstance(k, int):
+                    self.psi[j][k] = buffer_dict['tensor' + str(n)]
+                    n += 1
+
+        return scattering2d(input, self.J, self.L, self.subsample_fourier, self.pad, self.modulus, fft, cdgmm, unpad, self.phi, self.psi, self.max_order, self.M_padded, self.N_padded)
+
+    def forward(self, input):
+        """Forward pass of the scattering.
+
+        Parameters
+        ----------
+        input : tensor
+           tensor with 3 dimensions :math:`(B, C, M, N)` where :math:`(B, C)` are arbitrary.
+           :math:`B` typically is the batch size, whereas :math:`C` is the number of input channels.
+
+        Returns
+        -------
+        S : tensor
+           scattering of the input, a 4D tensor :math:`(B, C, D, Md, Nd)` where :math:`D` corresponds
+           to a new channel dimension and :math:`(Md, Nd)` are downsampled sizes by a factor :math:`2^J`.
+
+        """
+        if not torch.is_tensor(input):
+            raise (
+            TypeError('The input should be a torch.cuda.FloatTensor, a torch.FloatTensor or a torch.DoubleTensor'))
```

Comment:
Why not just say `The input should be a PyTorch Tensor` (or something like that)?